### PR TITLE
Patched explode query parameter support for SolidStart

### PR DESCRIPTION
**Pull request by davedbase**
The `@orval/solid-start` client generated broken query strings for array parameters declared with `explode: true` in an OpenAPI spec. Instead of repeating the key for each element (`country=AT&country=DE`), it produced comma-separated values (`country=AT,DE`) or otherwise incorrect output. The root cause was that `generateImplementation` used `new URLSearchParams(params as any)` — a simple cast that doesn't understand the OpenAPI `explode` directive.

Solution was to port the same explode-aware URL building logic that `@orval/fetch` already uses. Also adds additional tests to account for the edge case.

@melloware mind having a closer look to ensure this meets expectations?

Patches https://github.com/orval-labs/orval/issues/3053

## Summary by CodeRabbit

* **New Features**
  * Improved query-string handling in generated clients: supports exploded array parameters, mixed scalar/array cases, date-aware formatting, and path-level parameter merging/deduplication.

* **Tests**
  * Added comprehensive tests for query serialization: no-params, scalar, exploded arrays, non-exploded arrays, mixed scenarios, schema variants, and path-level merging.

* **Chores**
  * Added a new package dependency to the project manifest.

* **Breaking Changes**
  * Updated generator API surface impacting client generation behavior.

### CODE REVIEW

**Review by melloware — APPROVED** [review]
Yes this is what you would need. Looks good once the build is passing

File: packages/solid-start/src/index.ts [hunk 1]
```diff
@@ -164,9 +168,146 @@ const generateImplementation = (
...
+          | (OpenApiSchemaObject | OpenApiReferenceObject)[]
+          | undefined) ?? []
+      ).some(
+        (s) =>
+          resolveRef<OpenApiSchemaObject>(s, context).schema.type === 'array',
+      );
+
+    return (
+      parameterObject.in === 'query' && isArrayLike && parameterObject.explode
+    );
```
[the remark this answers is not in the document]
Comment (davedbase) [reply packages/solid-start/src/index.ts]: You're so annoying. But good catch. :p

**Comment by davedbase** [comment]
Wow, I went down a...rabbit hole (literally). Haha. CodeRabbit is aggressive but caught some interesting edge cases. Neat tool.

I'm good with this as is.

Ready to go @melloware 👍

**Review by melloware — APPROVED** [review]
